### DISCUSSION TITLE
CPU footprint pattern changed after upgrading from GKE 1.23 to 1.24

### DISCUSSION THREAD

**Discussion by Jojoooo1** [rabbitmq/rabbitmq-server#8142] [Other]
### Describe the bug

After upgrading from GKE 1.23 to 1.24, CPU use seems instable as shown in the picture below (the peak just started after the reboot on the new node). It is only happening for one node.

![Screenshot from 2023-05-11 15-38-34](https://github.com/rabbitmq/rabbitmq-server/assets/17641873/0fbca460-fdf5-42df-985a-5adf63b56c5e)

I am currently using bitnami helm chart 11.14.4 which uses RabbitMQ 3.11.15

### Reproduction steps

1. create cluster with GKE 1.23 and upgrade to GKE 1.24
2. follow rabbitmq resources.

### Expected behavior

CPU use should be stable.

**Comment by michaelklishin** [comment]
If you upgraded GKE and observed a difference, why do you expect the RabbitMQ core team to be looking into it?

We cannot suggest much with the amount of information provided. Consider collecting additional
metrics from your Kubernetes and RabbitMQ clusters first.

RabbitMQ docs provide a set of [recommendations for reducing CPU footprint of mostly idle clusters](https://rabbitmq.com/runtime.html#cpu-reduce-idle-usage).

**Comment by Jojoooo1** [comment]
Trying to investigate

![Screenshot from 2023-05-11 15-55-48](https://github.com/rabbitmq/rabbitmq-server/assets/17641873/df569fd4-32f8-45b1-9cbc-581b3ac99c8d)
![Screenshot from 2023-05-11 15-56-10](https://github.com/rabbitmq/rabbitmq-server/assets/17641873/b5887ef9-1a2a-4db0-b960-a1f3154732b0)

![Screenshot from 2023-05-11 15-58-38](https://github.com/rabbitmq/rabbitmq-server/assets/17641873/17bb889c-d464-49af-b437-741805cd7e4d)

**Reply by michaelklishin** [reply]
This looks like a node that has just booted, and that's it. A code loader module and async task work pool are the top contributors. Then application controller and the code (cache) server are number 3 and number 4.

This node has booted and is doing nothing, which is further confirmed by its grand memory footprint of 80 MiB.

**Reply by michaelklishin** [reply]
This is exactly the kind of environment that [these recommendations](https://rabbitmq.com/runtime.html#cpu-reduce-idle-usage) are for.

What in the new GKE version may have caused a CPU spike to 0.4 of one core, I won't speculate, we don't have any evidence that this condition is permanent.

**Comment by michaelklishin** [comment]
@Jojoooo1 are you sure you have any reasons to file this and ask for our attention? This node is seemingly doing nothing at all, both according to my observations above  https://github.com/rabbitmq/rabbitmq-server/discussions/8142?converting=1#discussioncomment-5876812, and the fact that 99% of runtime scheduler's time is spent sleeping.

**Comment by Jojoooo1** [comment]
I thought that It might have been related to GKE 1.24 sorry If got to the wrong direction. Having worked with the same cluster (and many upgrade) for more than 3 years and never had been any similar situations. This is why I got pretty surprise, even after starting to receive messages it was not getting stabilized.

**Comment by Jojoooo1** [comment]
@michaelklishin sorry for the convenience seems it might related to https://github.com/bitnami/charts/commit/83827ce340efd7f7203d1ca7789eb9205110a1e0
